datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java

```diff
@@ -68,12 +74,48 @@ public void onPartitionsRevoked(@NotNull Collection<TopicPartition> topicPartiti
         }
       }
     });
+
+    // Remove consumer offsets for partitions that have been revoked
```

Comment:
I think this code won't be called if `_enablePartitionAssignment` is true (since this doesn't use consumer group management). These `onPartitionAssigned()/onPartitionRevoked()` callbacks are Kafka callbacks invoked specifically for consumer group management only. For `onPartitionAssigned()`, we manually set up some of the state in `consumerSubscribe()` function for `_enablePartitionAssignment`. I don't see any handling there to revoke TopicPartitions which are no longer assigned for the partition tracker. We need to fix this for the topic manager related stuff in that function too.

Feel free to open a separate bug to handle this correctly (and feel free to take it up or assign it to me). Wanted to capture this potential gap that we may have (needs proper investigation to ensure there is no other code path which is taking care of this already) so that we don't forget to look into it.

Reply:
Will open a ticket to handle that case